### PR TITLE
Clean-up Folio migration cruft

### LINKED ISSUE #1982: Clean up temporary FOLIO migration code
Mostly from https://github.com/sul-dlss/sul-requests/pull/1674

E.g. some of the needed_date handling for the `Page` should be unnecessary.

### PR DESCRIPTION
**Pull request by mjgiarlo**
This change reverts changes made to support the ILS migration from Symphony to Folio

Fixes #1982

Includes:
* Remove `needed_date` concern from `Page` model
* Remove migration feature flag

### CODE REVIEW

File: app/controllers/mediated_pages_controller.rb [hunk 1]
```diff
@@ -23,7 +23,8 @@ def update_params
   end
 
   def validate_request_type
-    return if current_request.mediateable? || (Settings.features.migration && current_request.pageable? && params[:action] == 'update')
+    # TODO: Determine if the logic after the || should stick around
```
Comment (mjgiarlo) [inline app/controllers/mediated_pages_controller.rb:26]: Bringing attention to this question for reviewers.

File: app/controllers/mediated_pages_controller.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def update_params
   end
 
   def validate_request_type
-    return if current_request.mediateable? || (Settings.features.migration && current_request.pageable? && params[:action] == 'update')
+    return if current_request.mediateable? || (current_request.pageable? && params[:action] == 'update')
```
Comment (jcoyne) [inline app/controllers/mediated_pages_controller.rb:26]: Can't we pull out this second clause?  Wasn't `Settings.features.migration` being nil causing the rest of this to not be needed?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/controllers/mediated_pages_controller.rb (hunk 2 above)
Comment (mjgiarlo) [reply app/controllers/mediated_pages_controller.rb:26]: I had originally thought not, but tracing through the `blame` farther than I did at first, yes, I do believe the second clause was added to support the migration work. Change forthcoming. Thanks!

Comment (mjgiarlo) [reply app/controllers/mediated_pages_controller.rb]: Done.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**